Sources/AblyChat/RoomStatus.swift

```diff
@@ -31,3 +31,27 @@ public struct RoomStatusChange: Sendable {
         self.error = error
     }
 }
+
+internal actor DefaultRoomStatus: RoomStatus {
+    internal private(set) var current: RoomLifecycle = .initialized
+    // TODO: populate this (https://github.com/ably-labs/ably-chat-swift/issues/28)
+    internal private(set) var error: ARTErrorInfo?
+
+    // TODO: clean up old subscriptions (https://github.com/ably-labs/ably-chat-swift/issues/36)
+    private var subscriptions: [Subscription<RoomStatusChange>] = []
+
+    internal func onChange(bufferingPolicy: BufferingPolicy) -> Subscription<RoomStatusChange> {
```

Comment:
Why `onChange` returns a new subscription each time? I would expect it to call `emit` or something. Maybe it's a `subscribe`?

Reply:
I think this is an issue with the naming of this method — in JS this is a method which takes a listener and means "on status change, call this listener". When writing the public API in #9 I decided to call the method in Swift `subscribe` but changed it back to `onChange` after Andy asked about it (https://github.com/ably-labs/ably-chat-swift/pull/9#discussion_r1718445081). I don't really want to try and think of a better naming as part of this PR though, can we defer it?